### PR TITLE
fileservice: update AWS and MinIO sdk versions and refine tests

### PR DESCRIPTION
**Pull request by reusee**
## What type of PR is this?

- [ ] API-change
- [ ] BUG
- [x] Improvement
- [ ] Documentation
- [ ] Feature
- [ ] Test and CI
- [ ] Code Refactoring

## Which issue(s) this PR fixes:

issue #17001 

## What this PR does / why we need it:
update AWS S3 packages.

### CODE REVIEW

**Review by fengttt — COMMENTED** [review]
This one, are you sure?   Seems it is wrong -- use a value almost surely will be better than pointer?

**Comment by reusee** [comment]
> This one, are you sure? Seems it is wrong -- use a value almost surely will be better than pointer?

这是第三方库的函数参数类型变了，所以传递的参数类型也要变。语义没有变化。

File: pkg/fileservice/utils.go [hunk 1]
```diff
@@ -25,3 +25,11 @@ func ceilingDiv[T int | int64](n, by T) T {
 	}
 	return res + 1
 }
+
+func zeroToNil[T comparable](v T) *T {
+	var zero T
```
Comment (fengttt) [inline pkg/fileservice/utils.go:30]: does this return &v means a copy and alloc?   Should this arg use v *T?

**Review by zhangxu19830126 — APPROVED** [review]

File: pkg/fileservice/utils.go (hunk 1 above)
Comment (reusee) [reply pkg/fileservice/utils.go:30]: 这个函数会被内联，且没有分配，已经加了单测，如果有分配，就失败。

**Review by fengttt — APPROVED** [review]

**Pull request merged**